File: AlgoExpert/Stacks/Medium/Sunset Views.java
import java.util.*;

class Program {

  public ArrayList<Integer> sunsetViews(int[] buildings, String direction) {
    // Write your code here.
    Stack<Integer> stack = new Stack<>();

    int idx, step;
    if (direction.equals("EAST")) {
      idx = buildings.length - 1;
      step = -1;
    } else {
      idx = 0;
      step = 1;
    }

    int tallestBuildingSoFar = 0;
    while (idx >= 0 && idx < buildings.length) {
      if (buildings[idx] > tallestBuildingSoFar) {
        stack.push(idx);
        tallestBuildingSoFar = buildings[idx];
      }
      idx += step;
    }

    ArrayList<Integer> result = new ArrayList<>(stack);
    if (direction.equals("EAST"))
      Collections.reverse(result);
    
    return result;
  }
}

File: Python/AlgoExpert/Stacks/Medium/sunsetViews.py
def sunsetViews(buildings, direction):
    # Write your code here.
    if direction == 'EAST':
        start, stop, step = 0, len(buildings), 1
    else:
        start, stop, step = len(buildings) - 1, -1 , -1

    result = []
    for i in range(start, stop, step):
        print(i)
        while len(result) > 0:
            last_height = buildings[result[-1]]
            if last_height <= buildings[i]:
                result.pop()
            else:
                break
        result.append(i)

    if direction == 'WEST':
        return result[::-1]
        
    return result